### ISSUE TITLE
Dropdown text color matches background-color when inside .bg-primary

### ISSUE THREAD

**Issue by lidori** [twbs/bootstrap#16559]
I put a dropdown inside a panel, which I gave bg-primary to.
Since the bg-primary have white color for text, the dropdown inside is getting this and the values do not appear (white text on white background).
To avoid this I added my own style for dropdown with black text color. But I think it would be a good idea to add it in the bootstrap css for dropdowns.

```
<!DOCTYPE html>
<html>
<head>
<title>Test Dropdown, with workaround</title>
<!-- Latest compiled and minified CSS -->
<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/css/bootstrap.min.css">
<!-- Optional theme -->
<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/css/bootstrap-theme.min.css">
<style>
.dropdown{
	color: black
}
</style>
</head>
<body>
<div class="panel panel-default">
  <div class="panel-body bg-primary">
	<div class="dropdown">
		<button id="dLabel" type="button" class="btn btn-default" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
		Dropdown trigger
		<span class="caret"></span>
		</button>
		  <ul class="dropdown-menu" role="menu" aria-labelledby="dLabel">
			<li>a</li>
			<li>b</li>
			<li>c</li>
		  </ul>
		</div>
	</div>
</div>
<!-- Latest compiled and minified JavaScript -->
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.2/jquery.min.js"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
</body>
</html> 
```

**Comment by kkirsche** [comment]
Would you be able to post a jsbin example of this?

**Comment by cvrebert** [comment]
@mdo I presume this is WontFix per our guideline to keep the text color and background color utility classes simple and free of special cases?

**Comment by patrickhlauke** [comment]
@cvrebert @mdo I'd say it's more a fixup of dropdowns to explicitly set foreground color if they're also setting background color?

**Comment by lidori** [comment]
this is the original: http://jsbin.com/pivacodaya/1/edit?html,output
and this one with my workaround: http://jsbin.com/pufokabuxo/1/edit?html,output

**Comment by twbs-lmvtfy** [comment]
Hi @lidori!

You appear to have posted a live example (http://jsbin.com/pivacodaya/1/edit), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 4, column 1 thru line 4, column 16: Stray start tag ```` html ````.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by twbs-lmvtfy** [comment]
Hi @lidori!

You appear to have posted a live example (http://jsbin.com/pufokabuxo/1/edit), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 4, column 1 thru line 4, column 16: Stray start tag ```` html ````.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by cvrebert** [comment]
I believe Dropdown `<li>`s are currently required to contain `<a>`s (unless they're `.dropdown-header`s or `.divider`s). (Not that that's necessarily optimal.) Adding the missing `<a>`s seems to solve the color problem.

**Comment by cvrebert** [comment]
@lidori I took the liberty of making fixed JS Bins without the stray `<html>` tags and updating your comment accordingly.

**Comment by lidori** [comment]
I removed the double html tag from all the examples

**Comment by lidori** [comment]
Thanks, @cvrebert. I will use the <a>.
